### PR TITLE
User-defined escape function.

### PR DESCRIPTION
**Pull request by yakov-bakhmatov**
Tera is very nice template engine. But in my project I need custom escape function. So I added it to Tera, and I think it may be useful for other users. (Sorry for my English.)

### CODE REVIEW

File: src/tera.rs [hunk 1]
```diff
@@ -638,6 +675,22 @@ mod tests {
     use context::Context;
     use serde_json::{Map as JsonObject, Value as JsonValue};
 
+    fn escape_c_string(input: &str) -> String {
```
Comment (Keats) [inline src/tera.rs]: when would this be used? I think this doesn't need to be in Tera directly

Comment (yakov-bakhmatov) [reply src/tera.rs]: `escape_c_string` is in `tera::tests` module, so it is not visible in production code.

Comment (yakov-bakhmatov) [reply src/tera.rs]: I moved `escape_c_string` into test function.

Comment (Keats) [reply src/tera.rs]: I missed that ;( sorry

**Comment by Keats** [comment]
Looks good!
I'll merge it once I'm going to make a new release, shouldn't take too long

**Pull request merged**